api/lib/opentelemetry/context/propagation/propagation.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  class Context
+    module Propagation
+      # The Propagation class provides methods to inject and extract context
+      # to pass across process boundaries
+      class Propagation
+        HTTP_TRACE_CONTEXT_EXTRACTOR = Trace::Propagation::HttpTraceContextExtractor.new
```

Comment:
A small twist on this design from the otep was that propagators were modularized and returned their extractor/injector when invoked
```
  bagExtract, bagInject = Correlations::HTTPPropagator()
  traceExtract, traceInject = Tracer::B3Propagator()
```
It might read a little more elegantly if we don't have a single module for everything propagation

Reply:
This definitely needs some clean up. I have some ideas.

Reply:
This should be a little better now.